### PR TITLE
SPARK-6878 [CORE] Fix for sum on empty RDD fails with exception

### PR DESCRIPTION
**Pull request by erikvanoosten**

### CODE REVIEW

**Comment by srowen** [comment]
(Could you start the title with `SPARK-xxxx [CORE] ...` ? helps the merge messages and PR system. https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark)

File: core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala [hunk 1]
```diff
@@ -18,10 +18,17 @@
 package org.apache.spark.rdd
 
 import org.scalatest.FunSuite
+import org.scalatest.Matchers._
 
 import org.apache.spark._
 
 class DoubleRDDSuite extends FunSuite with SharedSparkContext {
+  test("sum") {
+    sc.parallelize(Seq.empty[Double]).sum() should be(0.0 +- 0.0001)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala]: I would just use `assert` here as int he rest of the file. Also I think you can assert exact equality in these cases. It really should be `3.0` in the last instance, precisely.

**Comment by srowen** [comment]
ok to test

File: core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala (hunk 1 above)
Comment (erikvanoosten) [inline core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala]: Experience thought me that exact equality with doubles is unreliable. Now I err on the safe side and always use inexact matchers in unit tests.

Comment (srowen) [inline core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala]: Doubles aren't unstable; `0.0 == 0.0` always. Yes i know what you mean but in these cases we can expect the result to exact to machine precision, always.

Comment (erikvanoosten) [inline core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala]: I didn't say unstable :)
Anyway, changed as requested.

**Comment by srowen** [comment]
LGTM pending tests, thank you.

**Pull request closed without merging**